framework/src/headers/ip/v6/icmp/packet_too_big.rs

```diff
@@ -0,0 +1,151 @@
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                             MTU                               |
+   +-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+-+
+   |                    As much of invoking packet                 |
+   +               as possible without the ICMPv6 packet           +
+   |               exceeding the minifmum IPv6 MTU [IPv6]           |
+
+   Type           2
+
+   Code           Set to 0 (zero) by the originator and ignored by the
+   receiver.
+
+   MTU            The Maximum Transmission Unit of the next-hop link.
+
+   A Packet Too Big MUST be sent by a router in response to a packet
+   that it cannot forward because the packet is larger than the MTU of
+   the outgoing link.  The information in this message is used as part
+   of the Path MTU Discovery process [PMTU].
+
+   Originating a Packet Too Big Message makes an exception to one of the
+   rules as to when to originate an ICMPv6 error message.  Unlike other
+   messages, it is sent in response to a packet received with an IPv6
+   multicast destination address, or with a link-layer multicast or
+   link-layer broadcast address.
+ */
+
+pub const IPV6_TOO_BIG_PAYLOAD_LEN: u16 = 1240; // MTU MIN - v6 size of 40;
+
+#[derive(Debug)]
+#[repr(C, packed)]
+pub struct Icmpv6PktTooBig<T>
+where
+    T: IpHeader,
+{
+    pub icmp: Icmpv6Header<T>,
+    mtu: u32,
+    _parent: PhantomData<T>,
+}
+
+impl<T> Default for Icmpv6PktTooBig<T>
+where
+    T: IpHeader,
+{
+    fn default() -> Icmpv6PktTooBig<T> {
+        Icmpv6PktTooBig {
+            icmp: Default::default(),
+            mtu: IPV6_MIN_MTU,
+            _parent: PhantomData,
+        }
+    }
+}
+
+impl<T> fmt::Display for Icmpv6PktTooBig<T>
+where
+    T: IpHeader,
+{
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(
+            f,
+            "msg_type: {} code: {} checksum: {}, mtu: {}",
+            self.msg_type().unwrap(),
+            self.code(),
+            self.checksum(),
+            self.mtu()
+        )
+    }
+}
+
+impl<T> EndOffset for Icmpv6PktTooBig<T>
+where
+    T: IpHeader,
+{
+    type PreviousHeader = T;
+
+    #[inline]
+    fn offset(&self) -> usize {
+        // ICMPv6 Header for Packet Too Big Msg (Type + Code + Checksum + MUT)
+        // is always 8 bytes: (8 + 8 + 16 + 32) / 8 = 8
+        8
+    }
+
+    #[inline]
+    fn size() -> usize {
+        // ICMPv6 Header is always 8 bytes so size = offset
+        8
+    }
+
+    #[inline]
+    fn payload_size(&self, _hint: usize) -> usize {
+        // We have to include as much of the offending original
+        // packet as possible, hence we will always max to MTU, not
+        // including the current header
+        (IPV6_MIN_MTU_CHECK as usize) - MacHeader::size() - Ipv6Header::size() - Self::size()
+    }
+
+    #[inline]
+    fn check_correct(&self, _prev: &T) -> bool {
+        true
+    }
+}
+
+impl<T> Icmpv6PktTooBig<T>
```

Comment:
what if we have a default trait that contains msg_type, code, checksum getter/setters and is used by all icmp message structs? it might cut down on some of the duplication. we might be able to do the same w/ offset, size methods as well?

Reply:
@scrollins I'm kinda using the generic icmp header that way within toobig, like we did w/ ext headers. Sadly, default traits with mainly default methods has some interesting specifics around the use of self, hence how the checksum work is defaulted here (e.g. https://github.com/williamofockham/NetBricks/pull/49/files#diff-73a4b0cfca69a2ab9bb8d86b63c14e8cR52 -- the getter and setters are part of the each trait impl). So, while we can do it, it won't help us get around anything. 

Also, the offset/size methods are part of endoffset, which is what everything traits off (and the values are different for each (and for every other icmp message that has additional fields).

Reply:
@scrollins one option is have a macro to generate some boilerplate tho, would be up for that eventually.

Reply:
yeah, a macro might be a good option here. just trying to cut down on code duplication

Reply:
@scrollins agreed. Think we can return to this though, once we get some cleanup time and properly plan it out.